tikvrpc/tikvrpc.go

```diff
@@ -813,7 +813,10 @@ func SetContext(req *Request, region *metapb.Region, peer *metapb.Peer) error {
 		ctx.RegionEpoch = region.RegionEpoch
 	}
 	ctx.Peer = peer
-	if !AttachContext(req, ctx) {
+
+	// Shallow copy the context to avoid concurrent modification.
+	copyCtx := req.Context
```

Comment:
Actually the `req` could still be modifed in different threads?

Reply:
I don't find data race in `req`.
Currently, the data race is between `req.Context` and `req.Req.Context`, since before this PR, `req.Req.Context` is a reference of `req.Context`.